File: RareDiseasePredicter/RareDiseasePredicter/Implementations/Disease.cs
using RareDiseasePredicter.Interfaces;
using System;
using System.Collections.Generic;
using System.Text;

namespace RareDiseasePredicter.Implementations {
    class Disease : IDisease {
        List<ISymptom> symptoms = new List<ISymptom>();
        string name = "";
        int ID = -1;
        private string description = null;
        private string href = null;

        public string Name {
            get { return name; }
            set { this.name = value; }
            }
        public string Description {
            get { return description; }
            set { this.description = value; }
            }
        public string Href {
            get { return href; }
            set { this.href = value; }
            }
        int IDisease.ID {
            get { return ID; }
            set { this.ID = value; }
            }

        ICollection<ISymptom> IDisease.symptoms {
            get { return symptoms; }
            set { this.symptoms = (List<ISymptom>)value; }
            }

        public Disease(string Name) { this.Name = Name; }

        public Disease(string name, List<ISymptom> symptoms) {
            this.name = name;
            this.symptoms = symptoms;
        }

        public Disease(string name, List<ISymptom> symptoms, int id, string description) {
            this.name = name;
            this.symptoms = symptoms;
            this.ID= id;
            this.description = description;
            }

        public Disease(string name, List<ISymptom> symptoms, int id, string description, string href) {
            this.name = name;
            this.symptoms = symptoms;
            this.ID= id;
            this.description = description;
            this.href = href;
            }

        public Disease() {

        }

        public bool AddSymptoms(ISymptom symptom) {
            symptoms.Add(symptom);
            return true;
        }

        public string GetName() {
            return name;
        }

        public ICollection<ISymptom> GetSymptoms() {
            return symptoms;
        }

        public bool SetName(string name) {
            this.name = name;
            return true;
        }

        public int GetID() {
            return this.ID;
        }

        public bool SetID(int ID) {
            this.ID = ID;
            return true;
        }
    }
}

File: RCPredicterCPP/RCPCPP/Disease.cpp
#include "Disease.h"


using namespace RareDiseaseCalculator;

Disease::Disease(std::string Name, int ID, std::vector<Symptom*> symptoms) {
	this->Name = Name;
	this->ID = ID;
	this->symptoms = symptoms;
}

Disease::~Disease() {
	delete this;
}

std::string Disease::getName() {
	return Name;
}

int Disease::getID() {
	return ID;
}

std::vector<Symptom*> Disease::getSymptoms() {
	return symptoms;
}

float Disease::getWeight() {
	return Weight;
}

std::vector<float> Disease::symptomWeight() {
	return symptomWeights;
}

Symptom::Symptom(std::string Name, int ID, std::vector<Region*> regions) {
	this->Name = Name;
	this->ID = ID;
	this->regions = regions;
}

RareDiseaseCalculator::Symptom::~Symptom() {
	delete this;
}

std::string Symptom::getName() {
	return Name;
}

int Symptom::getID() {
	return ID;
}

std::vector<Region*> Symptom::getRegions() {
	return regions;
}

Region::Region(std::string Name, int ID) {
	this->Name = Name;
	this->ID = ID;
}

RareDiseaseCalculator::Region::~Region() {
	delete this;
}

std::string Region::getName() {
	return Name;
}

int Region::getID() {
	return ID;
}

void Region::setName(std::string Name) {
	this->Name = Name;
}